tests/unit/test_protocol_errors.py

```diff
@@ -0,0 +1,105 @@
+"""
```

Comment:
We can't mandate the error codes for other implementations. These are specific to us.